### DISCUSSION TITLE
MSP2 over Crsf Telemety for PID lua Script like betaflight

### DISCUSSION THREAD

**Discussion by druckgott** [iNavFlight/inav#9830] [Ideas]
## Current Behavior
MSP2 is currently not Supported over CRSF Telemetry so we cant use lua script to change pids with transmitter

## Desired Behavior
MSP2 is supported over CRSF Telemetry so that we can use a first script and implement more and more things like on betaflight lua script

## Suggested Solution
It would be cool to have a Script like in betaflight to send PIDS and so on with LUA to change it on the Transmitter and do not need a App ore PC or anything else.

[klutvott123](https://github.com/klutvott123) have thxfully implementet mps2 for the lua script
https://github.com/betaflight/betaflight-tx-lua-scripts/issues/491

I have done a version which already starts with inav but only gets msp1 values which is currently not very usefull for PIDS.
https://github.com/druckgott/betaflight-tx-lua-scripts/tree/inav_version
I will later make a fork from betaflight script and change again all so that the commit history from betaflight is also availibe.

## Who does this impact? Who is this for?
Everybody who will use a lua script on Transmitter to change PIds without extra tool and not with any switches in INV which maybe not availble due to other functions

## Additional context
Would be realy cool if we can do this and then extend the lua script lick the betaflight lua script

**Comment by druckgott** [comment]
Did Start here no Compile error alrady, but Lua Script says Waiting for API and I do not know how to debug both (lua and inav itself)
https://github.com/iNavFlight/inav/pull/9528

**Comment by maxledfreak** [comment]
Hello @druckgott,

this feature sounds intriguing! While setting up via mobile phone has been a common approach in the past, it's not always the ideal solution. I've come across information suggesting that configuring it through switches and their connections might be a more efficient method. I will try this. 

Moreover, there's a LUA script available for iNav in general. Integrating it into this script could potentially enhance the setup process. From a usability standpoint, configuring it through a LUA script seems to be a superior option compared to using a mobile phone.

Have you had a successful run with your first PR?

@DzikuVx, could we consider adding this feature to the roadmap for upcomming iNav versions?

**Comment by MrD-RC** [comment]
Honestly, I find the adjustments so much more convenient. You can do it in flight. No need to land, remove your goggles, and look at your radio.

**Comment by druckgott** [comment]
It depends how you want to do it. For me in flight is not realy nice, I have do use so much buttons have to concentrate maybe something goes wrong and the wing is crashing which makes no sence for me. I want to land switch the value start again, get new log file ... and this is happening automaticaly if I use this script. I think this is depending on the people how they want to do it. Maybe for you it´s not nice but I think there are other people who find it nice. Maybe lets focus on adding good tings to inav ;-). I don´t want to discuss the preferences from people ;-).

I´m not a fan to have many swiches for on the rx to do maybe wrong things due to the many swiches. And there is a clear menü in lua script and I clearly know what I´m doing.

**Comment by RomanLut** [comment]
No so excited about tuning pids with lua script on my Jumper T-Lite screen :smiley:, but this feature would make wireless configurator communication possible https://github.com/iNavFlight/inav-configurator/issues/1616

**Comment by VanLaser** [comment]
Hello, not sure if this is the best place to ask, but is there any *compromise* solution available? By that, I mean the following: use the adjustments tab in INAV configurator to setup _changing_ of PIDs from radio controls, but also be able to at least _read_ the PID values using telemetry on the radio screen. Therefore, no need for video link/OSD. Is this easier to accomplish, or does it exist already in some Lua script based widgets/screens out there? And if not, any suggestion about what telemetry script would be the easier to hack on? Thanks!
